README.md

```diff
@@ -24,19 +24,21 @@ Quick links:
 FS2 is a streaming I/O library. The design goals are compositionality, expressiveness, resource safety, and speed. Here's a simple example of its use:
 
 ```scala
-import cats.effect.{ExitCode, IO, IOApp}
+import cats.effect.{ExitCode, IO, IOApp, Resource}
 import cats.implicits._
 import fs2.{io, text}
 import java.nio.file.Paths
 import java.util.concurrent.Executors
 import scala.concurrent.ExecutionContext
 
 object Converter extends IOApp {
-  def run(args: List[String]): IO[ExitCode] = {
+  private val blockingExecutionContextResource =
+    Resource.make(IO(ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))))(ec => IO(ec.shutdown()))
```

Comment:
_Maybe_ `Stream.bracket` here, since we're on topic?

Reply:
Or perhaps keep the Resource there, but `Stream.resource` rather than `use` later on

Reply:
Fine with me, but I prefer to use Resource if there's a single thing it'll emit, plus we'll have to carry it around the whole thing.

Reply:
Just saw your second comment, that still leaves the second issue...

Reply:
> Fine with me, but I prefer to use Resource if there's a single thing it'll emit,

Yeah having Resource there it's a good idea, since that's how we expose our apis as well now (e.g. `Socket`), but I'd rather showcase `Stream.resource` since we're at it, given that in many cases you want to use a Resource but stay in Stream

Reply:
```scala
object Converter extends IOApp {

  val blockingExecutionContext: Resource[IO, ExecutionContext] =
    Resource.make(IO(ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))))(ec => IO(ec.shutdown()))

  def converter: Stream[IO, Unit] = blockingExecutionContext.flatMap { blockingEC =>
    def fahrenheitToCelsius(f: Double): Double = (f - 32.0) * (5.0/9.0)

    io.file.readAll[IO](Paths.get("testdata/fahrenheit.txt"), blockingEC, 4096)
      .through(text.utf8Decode)
      .through(text.lines)
      .filter(s => !s.trim.isEmpty && !s.startsWith("//"))
      .map(line => fahrenheitToCelsius(line.toDouble).toString)
      .intersperse("\n")
      .through(text.utf8Encode)
      .through(io.file.writeAll(Paths.get("testdata/celsius.txt"), blockingEC))
  }


  override def run(args: List[String]): IO[ExitCode] =
    converter.compile.drain.as(ExitCode.Success)
}
```

Something like that perhaps?

Reply:
Done, I also changed the ReadmeExample tut doc - please take a look, especially at the ending part (in which I close the EC that's allocated somewhere in the middle of the page)

Reply:
Should be good now, I'll take a look tomorrow and see if I can come up with some good description of what's going on with the EC.